crowbar_framework/app/models/service_object.rb

```diff
@@ -919,6 +919,35 @@ def self.proposal_to_role(proposal, bc_name)
     RoleObject.new role
   end
 
+  #  we can speed-up the application of (n+1)th role if both(n,n+1)
+  #  roles are being applied on the same node.
+  #
+  #  eg. In our 2node deployment ceilometer{server,central} are always
+  #  applied on the same node, given that they have different priorities,
+  #  they are to be applied, one after the other.
+  #
+  #  In other words: it's actually reducing the number of times chef-client
+  #  is run rather than speeding up execution of any single run, by
+  #  merging batches together
+  #
+  #  a batch is [roles, nodes]
+  def mergebatches(batches)
```

Comment:
Yes, much better to have this as a separate method!

But as mentioned before, please stay consistent with the existing style, so this should be `merge_batches`.